File: ruby/f20.loop.rb
#!/usr/bin/ruby
# loop until user quits (begin...while construct)
begin
  print "Enter your name (quit to Exit): " # print prompt
  STDOUT.flush                             # flush buffer to show prompt
  answer = gets.chomp                      # get string input
 
  if answer != "quit"
      puts "Hello #{answer}"               # print if not exiting
  end
end while answer != "quit"

File: win_scripts/wsh.jscript/scripts/f20.loop.js
// conditional loop with do..while construct
do {
  // prompt user
  WScript.stdout.write("Enter your name (quit to Exit): ");
  answer = WScript.stdin.readline();         // get input
  if (answer != "quit")  
      WScript.echo("Hello " + answer + "!"); // output result if not exiting
} while (answer != "quit")
